project.go

```diff
@@ -43,12 +43,38 @@ func findProjectRoot(from string) (string, error) {
 type Project struct {
 	// AbsRoot is the absolute path to the root directory of the project.
 	AbsRoot string
+	// ResolvedAbsRoot is the resolved absolute path to the root directory of the project.
+	// If AbsRoot is not a symlink, then ResolvedAbsRoot should equal AbsRoot.
+	ResolvedAbsRoot string
 	// ImportRoot is the import path of the project's root directory.
 	ImportRoot gps.ProjectRoot
 	Manifest   *Manifest
 	Lock       *Lock // Optional
 }
 
+// SetRoot set the project AbsRoot. If the root is a symlink, it attempts to resolve it.
+func (p *Project) SetRoot(root string) error {
+	sym, err := fs.IsSymlink(root)
```

Comment:
What I was suggesting before would be this:

```go
func (p *Project) SetRoot(root string) error {
	rroot, err := filepath.EvalSymlinks(root)
	if err != nil {
		return err
	}
	
	p.ResolvedAbsRoot, p.AbsRoot = rroot, root
	return nil
}
```

I _think_ this would be equivalent. If `root` is not a symlink, then it's basically a no-op. If `root` is a symlink, then any error that would occur from `fs.IsSymlink()` are a subset of the errors from `filepath.EvalSymlinks()`.